tests/data/hierarchy_input.info

```diff
@@ -30,3 +30,17 @@ laplace
   reordering None
   n_refinements 5
 }
+
+;; Uncomment below for running ML
```

Comment:
Why not create `hierarchy_input_ml.info` input deck that is actually tested?

Reply:
Haven't put much thought into it.  I am not so keen on having two files that are almost the same.
That said, INFO format support include directives so I could move the ML specific parameters into a separate file and have a commented include directive here.  I don't think it makes things better though.

Reply:
Uncommenting these lines breaks matrix-free support (in `hierarrchy_driver`). I don't really like that. I would appreciate an approach where we can only tell the matrix-based version to use a different path.

Reply:
Fixed